storage/remote/write_handler.go

```diff
@@ -174,44 +140,34 @@ func (h *writeHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Deal with contentEncoding first.
-	var decompressed []byte
-
-	switch contentEncoding {
-	case "snappy":
-		decompressed, err = snappy.Decode(nil, body)
-		if err != nil {
-			level.Error(h.logger).Log("msg", "Error decoding remote write request", "err", err.Error())
-			http.Error(w, err.Error(), http.StatusBadRequest)
-			return
-		}
-	default:
-		level.Error(h.logger).Log("msg", "Error decoding remote write request", "err", "Unsupported Content-Encoding", "contentEncoding", contentEncoding)
-		// Return a 406 so that the client can choose a more appropriate protocol to use.
-		http.Error(w, "Unsupported Content-Encoding", http.StatusNotAcceptable)
+	decompressed, err := snappy.Decode(nil, body)
+	if err != nil {
+		// TODO(bwplotka): Add more context to responded error?
+		level.Error(h.logger).Log("msg", "Error decompressing remote write request", "err", err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
 
 	// Now we have a decompressed buffer we can unmarshal it.
-	// At this point we are happy with the version but need to check the encoding.
-	switch protoVer {
-	case RemoteWriteVersion1HeaderValue:
+	switch msg {
+	case config.RemoteWriteProtoMsgV1:
 		var req prompb.WriteRequest
 		if err := proto.Unmarshal(decompressed, &req); err != nil {
-			level.Error(h.logger).Log("msg", "Error decoding remote write request", "err", err.Error())
+			// TODO(bwplotka): Add more context to responded error?
+			level.Error(h.logger).Log("msg", "Error decoding remote write request", "protobuf_message", msg, "err", err.Error())
 			http.Error(w, err.Error(), http.StatusBadRequest)
 			return
 		}
 		err = h.write(r.Context(), &req)
-	case RemoteWriteVersion20HeaderValue:
-		// 2.0 request.
-		var reqMinStr writev2.Request
-		if err := proto.Unmarshal(decompressed, &reqMinStr); err != nil {
-			level.Error(h.logger).Log("msg", "Error decoding remote write request", "err", err.Error())
+	case config.RemoteWriteProtoMsgV2:
+		var req writev2.Request
+		if err := proto.Unmarshal(decompressed, &req); err != nil {
+			// TODO(bwplotka): Add more context to responded error?
+			level.Error(h.logger).Log("msg", "Error decoding remote write request", "protobuf_message", msg, "err", err.Error())
```

Comment:
we could at least include the proto message type, or make it `error decoding remote write request` and `error decoding v2 remote write request`?

Reply:
I added, but this question in TODO is more towards HTTP response. Will leave the TODO for now.